### PR TITLE
doc: add command options to README.md

### PR DESCRIPTION
**Pull request by gibm**
Documents options accepted by node-gyp.
Continuation of #880.

@bnoordhuis

### CODE REVIEW

**Comment by Fishrock123** [comment]
I feel like this is missing some options, also CC @bnoordhuis

**Comment by bnoordhuis** [comment]
Can you split it into two commits?  The new flag looks alright but a code change should not be part of a documentation commit.

Also, @Fishrock123 is right that there are plenty more options: `--arch`, `--python`, `--dist-url=`, etc.

**Comment by gibm** [comment]
Looking at `lib/node-gyp.js`, it looks like these are all options. I'll try to get examples added for all of them.
```
    help: Boolean     // everywhere
  , arch: String      // 'configure'
  , cafile: String    // 'install'
  , debug: Boolean    // 'build'
  , directory: String // bin
  , make: String      // 'build'
  , msvs_version: String // 'configure'
  , ensure: Boolean   // 'install'
  , solution: String  // 'build' (windows only)
  , proxy: String     // 'install'
  , nodedir: String   // 'configure'
  , loglevel: String  // everywhere
  , python: String    // 'configure'
  , 'dist-url': String // 'install'
  , 'tarball': String // 'install'
  , jobs: String      // 'build'
  , thin: String      // 'configure'
```

**Comment by gibm** [comment]
@bnoordhuis updated, PTAL

File: README.md [hunk 1]
```diff
@@ -144,13 +144,42 @@ Commands
...
+| `--thin=$thin`                    | Enable thin static libraries
+| `--arch=$arch`                    | Set target architecture (e.g. ia32)
+| `--tarball=$path`                 | Get headers from a local tarball
+| `--ensure`                        | Don't reinstall headers if already present
+| `--dist-url=$url`                 | Download header tarball from custom URL
+| `--proxy=$url`                    | Set HTTP proxy for downloading header tarball
+| `--cafile=$cafile`                | Override default CA chain (to download tarball)
+| `--nodedir=$path`                 | Set the path to the node binary
+| `--python=$path`                  | Set path to the python (2) binary
+| `--msys_version=$version`         | Set Visual Studio version (win)
```
Comment (bnoordhuis) [inline README.md]: Typo?  I think you mean `--msvs_version=$version`?

Comment (bnoordhuis) [inline README.md]: Typo?  I think you mean `--msvs_version=$version`?

File: README.md [hunk 2]
```diff
@@ -144,13 +144,42 @@ Commands
...
+|:----------------------------------|:------------------------------------------
+| `-j n`, `--jobs n`                | Run make in parallel
+| `--silly`, `--loglevel=silly`     | Log all progress to console
+| `--verbose`, `--loglevel=verbose` | Log most progress to console
+| `--silent`, `--loglevel=silent`   | Don't log anything to console
+| `debug`, `--debug`                | Make Debug build (default=Release)
+| `--release`, `--no-debug`         | Make Release build
+| `-C $dir`, `--directory=$dir`     | Run command in different directory
+| `--make=$make`                    | Override make command (e.g. gmake)
+| `--thin=$thin`                    | Enable thin static libraries
```
Comment (bnoordhuis) [inline README.md]: Maybe write this as `--thin=yes`?  Currently the only requirement is that it needs to be a non-empty string.

Comment (gibm) [inline README.md]: Yep, fixed.

**Comment by bnoordhuis** [comment]
Thanks Gibson, landed in ac29d23...4ee3132.

**Pull request closed without merging**